streamrouter/consul.py

```diff
@@ -216,6 +190,41 @@ def get_param(self, name):
         return self.params.get(name)
 
 
+class AwsServiceMixin:
+
+    get_aws_region_func = None
+
+    @cached_property
+    def aws_region(self):
+        assert self.raw_ptr is not None
+
+        return get_string(self.get_aws_region_func, self.raw_ptr)
+
+
+class CdnServiceMixin:
+
+    get_cdn_region_func = None
+    get_pop_func = None
+
+    @cached_property
+    def cdn_region(self):
+        assert self.raw_ptr is not None
+
+        region = self.get_cdn_region_func(self.raw_ptr)
+        if region == lib.CDN_REGION_EU:
+            return 'eu'
+        elif region == lib.CDN_REGION_NA:
+            return 'na'
+        else:
+            return None
+
+    @cached_property
```

Comment:
Nope. It isn't a verb here. It's an abbreviation of Point Of Presence. We use this abbreviation in our CDN to mark specific data centres. We don't use "data centre" because it'd ambiguous with Consul data centre property which is something else.

Reply:
YABA